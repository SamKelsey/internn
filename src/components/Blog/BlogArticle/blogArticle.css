.section-article {
  width: 70%;
  min-width: 350px;
  max-width: 700px;
  margin: 30px auto;
  border-radius: 10px;
  background-color: #f4f2f3;
  color: #595959;
  -webkit-box-shadow: rgba(0, 0, 0, 0.35) 0px 2px 4px;
          box-shadow: rgba(0, 0, 0, 0.35) 0px 2px 4px;
  overflow: hidden;
  font-family: "Montserrat", sans-serif;
}

.section-article .header-wrapper {
  position: relative;
  width: 100%;
}

.section-article .header-wrapper img {
  width: 100%;
  -webkit-filter: brightness(30%);
          filter: brightness(30%);
}

.section-article .header-wrapper h1 {
  position: absolute;
  top: 0;
  left: 0;
  color: rgba(244, 242, 243, 0.7);
  font-size: 3.5rem;
  font-weight: lighter;
  padding: 3%;
  margin: 0;
}

.section-article .header-wrapper h1 span {
  font-weight: 900;
}

.section-article .content-wrapper {
  text-align: justify;
  padding: 0 3%;
}

.section-article .content-wrapper h2 span {
  color: #f07474;
}

@media (max-width: 720px) {
  .section-article .header-wrapper h1 {
    font-size: 2.5rem;
  }
}

@media (max-width: 600px) {
  .section-article .header-wrapper h1 {
    font-size: 2rem;
  }
}
/*# sourceMappingURL=blogArticle.css.map */