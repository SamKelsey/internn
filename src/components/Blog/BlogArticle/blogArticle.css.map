{
    "version": 3,
    "mappings": "AAEA,AAAA,gBAAgB,CAAC;EACf,KAAK,EAAE,GAAG;EACV,SAAS,EAAE,KAAK;EAChB,SAAS,EAAE,KAAK;EAChB,MAAM,EAAE,SAAS;EACjB,aAAa,EAAE,IAAI;EACnB,gBAAgB,ECNP,OAAO;EDOhB,KAAK,ECLG,OAAe;EDMvB,UAAU,EAAE,mBAAmB,CAAC,GAAG,CAAC,GAAG,CAAC,GAAG;EAC3C,QAAQ,EAAE,MAAM;CAsDjB;;AA/DD,AAWE,gBAXc,CAWd,eAAe,CAAC;EACd,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,IAAI;CAsBZ;;AAnCH,AAeI,gBAfY,CAWd,eAAe,CAIb,GAAG,CAAC;EACF,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,eAAe;CACxB;;AAlBL,AAoBI,gBApBY,CAWd,eAAe,CASb,EAAE,CAAC;EACD,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,CAAC;EACN,IAAI,EAAE,CAAC;EAEP,KAAK,ECxBM,wBAAwB;EDyBnC,SAAS,EAAE,MAAM;EACjB,WAAW,EAAE,OAAO;EACpB,OAAO,EAAE,EAAE;EACX,MAAM,EAAE,CAAC;CAKV;;AAlCL,AA+BM,gBA/BU,CAWd,eAAe,CASb,EAAE,CAWA,IAAI,CAAC;EACH,WAAW,EAAE,GAAG;CACjB;;AAjCP,AAqCE,gBArCc,CAqCd,gBAAgB,CAAC;EACf,UAAU,EAAE,OAAO;EACnB,OAAO,EAAE,IAAI;CAOd;;AA9CH,AA0CM,gBA1CU,CAqCd,gBAAgB,CAId,EAAE,CACA,IAAI,CAAC;EACH,KAAK,EC7CJ,OAAO;CD8CT;;AAIL,MAAM,EAAE,SAAS,EAAE,KAAK;EAhD1B,AAkDM,gBAlDU,CAiDZ,eAAe,CACb,EAAE,CAAC;IACD,SAAS,EAAE,MAAM;GAClB;;;AAIL,MAAM,EAAE,SAAS,EAAE,KAAK;EAxD1B,AA0DM,gBA1DU,CAyDZ,eAAe,CACb,EAAE,CAAC;IACD,SAAS,EAAE,IAAI;GAChB",
    "sources": [
        "blogArticle.scss",
        "../../_styleVariables.scss"
    ],
    "names": [],
    "file": "blogArticle.css"
}