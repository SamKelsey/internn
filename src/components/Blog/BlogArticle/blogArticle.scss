@import "../../styleVariables";

.section-article {
  width: 70%;
  min-width: 350px;
  max-width: 700px;
  margin: 30px auto;
  border-radius: 10px;
  background-color: $sigWhite;
  color: $sigGrey;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 2px 4px;
  overflow: hidden;

  .header-wrapper {
    position: relative;
    width: 100%;

    img {
      width: 100%;
      filter: brightness(30%);
    }

    h1 {
      position: absolute;
      top: 0;
      left: 0;

      color: $sigWhiteOpaque;
      font-size: 3.5rem;
      font-weight: lighter;
      padding: 3%;
      margin: 0;

      span {
        font-weight: 900;
      }
    }
  }

  .content-wrapper {
    text-align: justify;
    padding: 0 3%;

    h2 {
      span {
        color: $sigRed;
      }
    }
  }

  @media (max-width: 720px) {
    .header-wrapper {
      h1 {
        font-size: 2.5rem;
      }
    }
  }

  @media (max-width: 600px) {
    .header-wrapper {
      h1 {
        font-size: 2rem;
      }
    }
  }
}
