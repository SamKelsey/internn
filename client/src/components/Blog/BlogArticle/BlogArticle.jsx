import React, { Component } from "react";
import { useParams } from "react-router-dom";

import getBlogInfo from "../../utilities/getBlogInfo";
import cardImage from "../../../images/IMG_0307-min.jpg";

import "./blogArticle.css";

const renderBody = (body) => {
  return body.map(({ header, content }) => {
    return (
      <React.Fragment>
        <h2>{header}</h2>
        <p>{content}</p>
      </React.Fragment>
    );
  });
};

const BlogArticle = () => {
  let { articleId } = useParams();
  const { title, subtitle, intro, body } = getBlogInfo(articleId);

  console.log(body);

  return (
    <div className="section-article">
      <div className="header-wrapper">
        <img src={cardImage} alt="Article" />
        <h1>
          <span>{title}:</span> {subtitle}
        </h1>
      </div>
      <div className="content-wrapper">
        <p>{intro}</p>
        {renderBody(body)}
      </div>
    </div>
  );
};

export default BlogArticle;
