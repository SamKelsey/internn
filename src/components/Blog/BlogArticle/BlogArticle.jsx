import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import { Helmet } from "react-helmet";

import getBlogInfo from "../../utilities/getBlogInfo";
import cardImage from "../../../images/IMG_0307.JPG";

import "./blogArticle.css";

const BlogArticle = () => {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);
  let { articleId } = useParams();
  const { title, subtitle, intro, body } = getBlogInfo(articleId);

  const renderBody = (body) => {
    return body.map(({ header, content, isNumbered }, index) => {
      return (
        <div key={index}>
          <h2>
            {isNumbered && <span>{index + 1}. </span>}
            {header}
          </h2>
          <p>{content}</p>
        </div>
      );
    });
  };
  return (
    <div className="section-article">
      <Helmet>
        <link
          rel="canonical"
          href="https://www.internn.co.uk/blog/prepare-your-property"
        />
      </Helmet>
      <div className="header-wrapper">
        <img src={cardImage} alt="Article" />
        <h1>
          <span>{title}:</span> {subtitle}
        </h1>
      </div>
      <div className="content-wrapper">
        <p>{intro}</p>
        {renderBody(body)}
      </div>
    </div>
  );
};

export default BlogArticle;
