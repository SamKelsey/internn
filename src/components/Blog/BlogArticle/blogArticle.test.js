import BlogArticle from "./BlogArticle";

import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

const sampleData = {
  title: "Sample title",
  subtitle: "Here is a subtitle",
  intro: "Sample intro",
  body: [
    {
      header: "Set the scene.",
      content: "Some sample content 1",
      isNumbered: true,
    },
    {
      header: "Remove the personal.",
      content: "Some sample content 2",
      isNumbered: false,
    },
  ],
};

jest.mock("../../utilities/getBlogInfo", () => ({
  __esModule: true,
  default: () => sampleData,
}));

test("Renders correctly.", () => {
  const { baseElement } = render(
    <MemoryRouter initialEntries={["/blog/prepare-your-property"]}>
      <BlogArticle />
    </MemoryRouter>
  );

  expect(baseElement).toMatchSnapshot();
});
