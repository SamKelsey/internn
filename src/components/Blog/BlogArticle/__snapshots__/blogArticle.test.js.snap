// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<body>
  <div>
    <div
      class="section-article"
    >
      <div
        class="header-wrapper"
      >
        <img
          alt="Article"
          src="IMG_0307.JPG"
        />
        <h1>
          <span>
            Sample title
            :
          </span>
           
          Here is a subtitle
        </h1>
      </div>
      <div
        class="content-wrapper"
      >
        <p>
          Sample intro
        </p>
        <div>
          <h2>
            <span>
              1
              . 
            </span>
            Set the scene.
          </h2>
          <p>
            Some sample content 1
          </p>
        </div>
        <div>
          <h2>
            Remove the personal.
          </h2>
          <p>
            Some sample content 2
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
